app/controllers/pages_controller.rb

```diff
@@ -0,0 +1,31 @@
+class PagesController < ApplicationController
+  before_action :set_webbook, only: [:show]
+  before_action :validate_read_webbook, only: [:show]
+
+  def show
+    @webbook = Webbook.find_by(id: params[:webbook_id])
+    @page = Page.where(webbook_id: params[:webbook_id]).find_by(id: params[:id])
+  end
+
+  private
+
+  def page_params
+    params.require(:page)
+          .permit(:title, :content, :page_number)
+  end
+
+  def user_has_book?
+    current_user.purchase_history
+                .purchase_history_webbooks
+                .find_by(webbook_id: params[:webbook_id])
+                .present?
```

Comment:
find_by.present? であれば、exists? でよさそう

```suggestion
                .exists?(webbook: @webbook)
```